docs: kube access multiple roles corrections (#39926)
gravitational/teleport: docs/pages/kubernetes-access/controls.mdx

headers later on.

Next, the Kubernetes Service checks each of the user's roles for `spec.deny`
conditions that might apply to the request:

- For `spec.deny.kubernetes_labels`, if any labels match the cluster's labels 

>   for the request, the Service denies the request before it is made to the 

Reviewer: ```suggestion
  for the request, the Teleport Kubernetes Service denies the request before it is made to the 
```

  Kubernetes API. 

> - For `spec.deny.kubernetes_resources`, if the resource matches a `deny` condition, 

Reviewer 2: In this case, the behavior is that you can't do the request. We won't change the `kubernetes_groups` or users.
If you have `kubernetes_users|groups` deny conditions, you can't use them at all

> - For `spec.deny.kubernetes_resources`, if the resource matches a `deny` condition, 
>   the associated `deny.kubernetes_groups` and `deny.kubernetes_users` are 
>   removed from the previously developed list of allowed RBAC principals from 
>   `spec.allow`. 

Reviewer: ```suggestion
- For `spec.deny.kubernetes_resources`, if the resource in the request matches a `deny` condition, 
  the Teleport Kubernetes Service removes the associated `deny.kubernetes_groups` and `deny.kubernetes_users` from the list of allowed RBAC principals it created from `spec.allow`. 
```
Using active voice

> Please note that a match in `spec.deny.kubernetes_labels` leads to an immediate 
> denial without a chance to adjust the user's impersonation data based on allow 
> rules of other roles they may have. On the other hand, a match in 
> `spec.deny.kubernetes_resources` only affects the user's RBAC principals but does 
> not result in an immediate denial.

Reviewer: ```suggestion
In other words, a match in `spec.deny.kubernetes_labels` leads to an immediate 
denial of the user's request. A match in 
`spec.deny.kubernetes_resources` only affects the user's RBAC principals but does 
not result in an immediate denial.
```
Trimming this a bit and removing "please".

> not result in an immediate denial.

Reviewer 2: Same as before. 
If a role specified `deny.kubernetes_groups=system:masters` and you have other role that grants `allow.kubernetes_groups=["system:masters", "view"]` the following happens:
- kubectl get pods -> uses `view` group. We collect `system:masters` and `view` from the allow clause, and after that we exclude the `system:masters`
- kubectl get pods --as user --as-group= view -> uses `view` group. We collect `system:masters` and `view` from the allow clause, and after that we exclude the `system:masters`. We compare against the request headers and the request goes through
- kubectl get pods --as user --as-group=system:masters -> request is denied. The user can't use the system:masters group
- kubectl get pods --as user --as-group=system:masters --as-group=view -> request is denied. The user can't use the system:masters group
